server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/Serializer.scala

```diff
@@ -156,9 +158,11 @@ class Serializer(mailboxIdFactory: MailboxId.Factory) {
   private implicit val maySubmitWrites: Writes[MaySubmit] = Json.valueWrites[MaySubmit]
   private implicit val mailboxRightsWrites: Writes[MailboxRights] = Json.writes[MailboxRights]
 
-  private implicit val personalNamespaceWrites: Writes[PersonalNamespace] = namespace => JsString("Personal")
-  private implicit val delegatedNamespaceWrites: Writes[DelegatedNamespace] = namespace => JsString(s"Delegated[${namespace.owner.asString}]")
-  private implicit val mailboxNamespaceWrites: Writes[MailboxNamespace] = Json.writes[MailboxNamespace]
+  private implicit val mailboxNamespaceWrites: Writes[MailboxNamespace] = {
+    case personal: PersonalNamespace => JsString("Personal")
+    case delegated: DelegatedNamespace => JsString(s"Delegated[${delegated.owner.asString}]")
+    case _ => JsNull
```

Comment:
It should break IMO, so I would return an JsError.

Reply:
It requires to return a `JsValue`, cannot return a `JsError`?

Reply:
Not even sure this case is needed btw?

Reply:
Yes with an algebrical object (sealed trait) we should get ride of this.

Reply:
I don't really know but JsNull doesn't seem correct. Can you check how we should handle that?

Reply:
Ok after checking a bit, the definition of a sealed trait in Scala seems to be:

> Sealed provides exhaustive checking for our application. Exhaustive checking allows to check that all members of a sealed trait must be declared in the same file as of the source file. That means that all the possible known members of a trait that must be included are known by the compiler in advance. So this gives us advantage to prevent mistakes in our code.

Which means you can't have other types of `MailboxNamespace` being declared somewhere else, so we know we only have those 2 cases to manage. 

Thus I agree with @chibenwa, we don't need the `case _` anymore with a sealed trait.

Would you agree @mbaechler ?